metric_learn/base_metric.py

```diff
@@ -458,9 +460,14 @@ def score(self, quadruplets, y=None):
     score : float
       The quadruplets score.
     """
-    quadruplets = check_input(quadruplets, y, type_of_inputs='tuples',
-                              preprocessor=self.preprocessor_,
-                              estimator=self, tuple_size=self._tuple_size)
+    checked_input = check_input(quadruplets, y, type_of_inputs='tuples',
+                                preprocessor=self.preprocessor_,
+                                estimator=self, tuple_size=self._tuple_size)
+    # checked_input will be of the form `(checked_quadruplets, checked_y)` if
+    # `y` is not None, or just `checked_quadruplets` if `y` is None
+    quadruplets = checked_input if y is None else checked_input[0]
```

Comment:
I find this a bit ugly, but I couldn't find another way to do it. Maybe refactor `check_input` with two functions like `check_X_y` and `check_array` ? Or we can just keep it as it is for now